Doc/#1209 - #1250 - Add documentation page for each application and page template (#1263)
Avaiga/taipy-doc: docs/tp_templates/pages/data_management_page.md

  out-of-the-box.</figcaption>
</figure>

This page template offers several key benefits:

- **Ease of Use**: The page template is designed to be user-friendly, with a simple CLI
    interface that guides developers through the page creation process on top of existing
    application. The pages then can be easily customized and plugged into the application.
- **Comprehensive Features**: The template provides a best-practice page scaffold for a
    data node management page, which supports a wide range of functionalities, providing a
    comprehensive solution for various use cases.
- **Customization**: The template provides high flexibility and customization options on page
    creation, allowing developers to tailor the page to meet specific requirements and use cases.

# How to create a page

To create the page from the data management page template, change to the folder of the application in
which you want to create the page and run `taipy create --page data_management` from the CLI. Then
answer a few questions to customize your page.

```console
$ taipy create --page data_management
[1/3] Page title (data_node_management_page):
[2/3] The folder that contains the pages (pages):

> [3/3] There are binding variables in the page.

Reviewer: Once and for all the 50 times this appears in this PR.
"binding variable"  is not a thing.
We can discuss this sticky problem

If the variables need to be imported from a different page,
please specify the module and the variable name separated by a space
(e.g. ..main selected_data_node):
selected_data_node ():

New Taipy page has been created at ./pages/data_node_management_page

> The "selected_data_node" binding variables have been created for the page

Reviewer: ```suggestion
The *selected_data_node* variable has been created for the page
```
I don't understand the plural. And in the rest of the section also.

You can replace the binding variables with your own variables in the page content.
Please import the new page in your main application to use it.
```

??? info "Default answers"

    In the CLI, the default answer for each question is displayed in the square brackets.
    You can provide an answer or press Enter to use the default value.

Each question in the CLI corresponds to a specific aspect of the page. The following
sections describe each question in detail.

## 1. Page title

- Specifies the title of the page.

> - The default value is "data_node_management_page".

Reviewer: Really? That certainly does not look like a page title...

## 2. The folder that contains the pages

- Specifies the path of the folder that contains the generated page.
- The path is relative to the current working directory. The folder will be created if not exist.
- The default value is "pages".

## 3. Binding variable for "selected_data_node"

> - Specifies the binding variable for the "selected_data_node" in the page content.

Reviewer: What do you mean "the "selected_data_node""??

> - The binding variable for the "selected_data_node" is used to store the data node selected by the

Reviewer: ?
Do you mean:
"The *selected_data_node* variable stores the selected data node."?

user in the page. For more information, please refer to
[Binding variables](../../userman/gui/binding.md).

> - By default, the binding variable is created in the page. You can import this variable to another

Reviewer: ```suggestion
- By default, this variable is defined in the page. You can import this variable to another
```

    page to share the data node selected by the user.

> - If there is already a binding variable with the same purpose from a different page, please

Reviewer: ```suggestion
- If there is already a variable for the same purpose from a different page, please
```

>     specify the module and the variable name separated by a space. It will be imported automatically

Reviewer: Specify how? An example maybe?

    to the data node management page.

# Page description

The generated page has the following folder structure:

```plaintext
pages/
├──── data_node_management_page/
│   ├──── __init__.py
│   └──── data_node_management_page.py
│
└──── __init__.py
```

Your page's folder structure may vary depending on the answers you provided during the
creation process. Here is a brief overview of the key components:

- *pages/*: The folder that contains the page specified in the second question.
    - *data_node_management_page/data_node_management_page.py* contains the layout and content of

>     the page. It includes a sidebar with the

Reviewer: ```suggestion
    the page. It includes a sidebar holding a
```

    [data_node_selector](../../refmans/gui/viselements/corelements/data_node_selector.md)

>     visual element, and the main content area with the

Reviewer: ```suggestion
    visual element, and a main content area showing a
```

    [data_node](../../refmans/gui/viselements/corelements/data_node.md) visual element.

>     - *\_\_init\_\_.py* is the file that imports the newly created page in the `pages` package.

Reviewer: ```suggestion
    - *\_\_init\_\_.py* is the file that exports the page from the `pages` package.
```

# Customizing the page

Everything in the generated page can be updated to precisely fit your specific requirements.

> The grid layout of the page can be customized via the

Reviewer: ```suggestion
The grid layout of the page can be customized via a
```

> [layout](../../refmans/gui/viselements/generic/layout.md) visual element.

Reviewer: ```suggestion
[layout](../../refmans/gui/viselements/generic/layout.md) block.
```

You can also customize the
[data_node_selector](../../refmans/gui/viselements/corelements/data_node_selector.md)

> and [data_node](../../refmans/gui/viselements/corelements/data_node.md) visual elements which

Reviewer: ```suggestion
and [data_node](../../refmans/gui/viselements/corelements/data_node.md) controls which
```

allow the user to select and manage the data nodes respectively.

You can also add more content to the page such as descriptions, or other
[visual elements](../../refmans/gui/viselements/index.md) provided by Taipy to enrich the
user experience.

# How to use the page

To use the page in your application, import the page in the main application file along with other
pages and add it to the page list given to the `Gui^` service.

```python title="main.py"
from taipy import Gui
from pages import root_page, data_node_management_page

pages = {
    "/": root_page,
    "data": data_node_management_page,
}

if __name__ == "__main__":
    gui = Gui(pages=pages)
    gui.run()
```

> You can also explicitly import the binding variable `selected_data_node`  from the newly created

Reviewer: ```suggestion
You can also explicitly import the variable *selected_data_node*  from the newly created
```
